### ISSUE TITLE
Modularize all the sub generator to follow yeoman recommended priority names

### ISSUE THREAD

**Issue by deepu105** [jhipster/generator-jhipster#2603]
Modularize all the sub generator to follow yeoman recommended priority names

Main generator restructuring is done here https://github.com/jhipster/generator-jhipster/pull/2595
Entity gene is restructured here https://github.com/jhipster/generator-jhipster/pull/2569

We need to do the same for language, module, service, and all the cloud subs

**Edit:** The restructuring is complete for all core tasks. Pending sub generators are below,
For these the `console.log` needs to be replaced with `this.log` as well as per #2567 

- [x] * ~~AWS~~ done
- [x] * cloudfoundry
- [x] * heroku
- [x] * ~~openshift~~ removed

**Comment by deepu105** [comment]
I'll do module, language and service. Any volunteer for the openshift, cloudfoundry, aws and heroku subs?

@wuethrich44 do you wanna do AWS sub restructure?

**Comment by wuethrich44** [comment]
@deepu105 I can do that for sure. Could you send me a link where I will see the recommended changes?

**Comment by deepu105** [comment]
refer the run loop part here http://yeoman.io/authoring/running-context.html
you can also refer entity/index.js or app/index.js I have already
restructured these

On Tue, Jan 12, 2016 at 8:39 PM, Fabian Wüthrich <notifications@github.com>
wrote:

> @deepu105 <https://github.com/deepu105> I can do that for sure. Could you
> send me a link where I will see the recommended changes?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jhipster/generator-jhipster/issues/2603#issuecomment-170899426>
> .
>



-- 
thank you,
Deepu.k.s

**Comment by wuethrich44** [comment]
@deepu105 I will implement this names in the next few days. Just three short questions:

* Should I do the build in `configuring`?
* and where should I put the other deployment steps `default` oder `writing`?
* Is the new error handling done with `this.env.error(chalk.red())` instead of the following method?

```javascript
AwsGenerator.prototype._errorHandling = function _errorHandling(err) {
    if (err) {
        this.abort = true;
        this.log.error(err.message);
    }
};
```

**Comment by deepu105** [comment]
@wuethrich44 

1. I think build should be in default
2. these should be in default and if there is any file write then do it in writing instaed
3. yeoman recommends `this.env.error` and I feel it is cleaner as well optimal than doing custom logic

**Comment by deepu105** [comment]
Any volunteers for cloudfoundry and heroku?

**Comment by deepu105** [comment]
@phouverneyuff I guess you use Heroku, would you be interested in doing a restructuring for the heroku sub generator?

**Comment by jdubois** [comment]
You need to do this with @jkutner : he's working for Heroku and he is the one doing this sub-generator

**Comment by phouverneyuff** [comment]
@deepu105 @jdubois I'm changing the code to fix a bug and I can do it. In the PR #2884 I'll remove the generator and move all code to a new module. I just need some time to learn about modules.

@jkutner You can contact me in PR #2884 or in the issue #2876 .

**Comment by deepu105** [comment]
I guess you got confused, this is not to move this to a module, this thread
is to make the sub generator code in line with latest yeoman recommended
templates,  look at the aws sub generator or language sub generator for
example

**Comment by phouverneyuff** [comment]
@deepu105 OK

**Comment by wmarques** [comment]
I'm actually refactoring the cloudfoundy subgenerator.

**Comment by wmarques** [comment]
I've done cloudfoundry and heroku subgen refactoring, I tried to be consistent with AWS file, can you check those commits : [Heroku](https://github.com/wmarques/generator-jhipster/commit/bd5290b5c821b5579adc2c86f165749d69634719), [Cloudfoundry](https://github.com/wmarques/generator-jhipster/commit/f0bf20e2a73c964949b0c6dd7b5e00d1a2ecf252) and tell me if it's ok for a PR ?

**Comment by jkutner** [comment]
@wmarques taking a look now

**Comment by jkutner** [comment]
@wmarques :+1: looks great. tested. worked.

**Comment by jkutner** [comment]
@wmarques i should clarify, I only reviewed and tested the Heroku part.

**Comment by wmarques** [comment]
@jkutner Thanks for your review ! I'm waiting for Cloudfoundry validation, I tested with my trial account and it worked

**Comment by deepu105** [comment]
@wmarques LGTM plz submit PR

**Comment by deepu105** [comment]
closed via #3040
